### DISCUSSION TITLE
FakeTPM or FakePluto features based on FakeSMC

### DISCUSSION THREAD

**Discussion by SelimCelen** [CloverHackyColor/CloverBootloader#505] [Ideas]
Hello i want to know if it's possible to implement
Tpm and Pluton security chip emulation like FakeSmc 
but as efi stub not kext.

**Comment by SergeySlice** [comment]
To create emulation one should know full description how it works.

**Comment by DMahalko** [comment]
The specification for how TPM works is available from the developers of the TPM standard.

https://trustedcomputinggroup.org/resource/tpm-library-specification/

6.12 TPM_CAP (Capabilities)
The TPM_CAP values are used in TPM2_GetCapability() to select the type of the value to be returned.

"Hello I'm TPM 2.0, but I'm not capable of anything, I'm just a fake stub so you won't refuse to run outright. Don't ask me to do anything for you. Kthxbye."
